packages/react/src/components/TextArea/__tests__/TextArea-test.js

```diff
@@ -309,4 +310,146 @@ describe('TextArea', () => {
+
+      fireEvent.change(screen.getByRole('textbox'), {
+        target: {
+          value: 'one two three four five six seven eight nine ten',
+        },
+      });
+
+      await waitFor(() => {
+        expect(screen.getByText('10/10')).toBeInTheDocument();
+      });
+
+      await waitFor(() => {
+        expect(screen.getByRole('textbox')).toHaveValue(
+          'one two three four five six seven eight nine ten'
+        );
+      });
+    });
+
+    it('should correctly decrease word count', async () => {
+      render(
+        <TextArea
+          id="input-1"
+          labelText="TextArea label"
+          enableCounter
+          maxCount={10}
+          counterMode="word"
+          defaultValue="one two three four"
+        />
+      );
+
+      // by default should show 4
+      expect(screen.getByText('4/10')).toBeInTheDocument();
+
+      fireEvent.change(screen.getByRole('textbox'), {
+        target: {
+          value: 'one two three',
+        },
+      });
+
+      await waitFor(() => {
+        expect(screen.getByText('3/10')).toBeInTheDocument();
+      });
+
+      await waitFor(() => {
+        expect(screen.getByRole('textbox')).toHaveValue('one two three');
+      });
+    });
+
+    it('should not trim words when enableCounter is disabled and then enabled', async () => {
+      const { rerender } = render(
+        <TextArea
+          id="input-1"
+          labelText="TextArea label"
+          enableCounter={false}
+          maxCount={5}
+          counterMode="word"
+          defaultValue="one two three four five"
+        />
+      );
+
+      fireEvent.change(screen.getByRole('textbox'), {
+        target: {
+          value: 'one two three four five six seven eight',
+        },
+      });
+
+      await waitFor(() => {
+        expect(screen.getByRole('textbox')).toHaveValue(
+          'one two three four five six seven eight'
+        );
+      });
+
+      rerender(
+        <TextArea
+          id="input-1"
+          labelText="TextArea label"
+          enableCounter={true}
+          maxCount={5}
+          counterMode="word"
+          // defaultValue="one two three four five"
+        />
+      );
+
+      await waitFor(() => {
+        expect(screen.getByRole('textbox')).toHaveValue(
+          'one two three four five six seven eight'
+        );
+      });
+    });
+    it('should trim words when text larger than max limit is pasted', async () => {
+      render(
+        <TextArea
+          id="input-1"
+          labelText="TextArea label"
+          enableCounter={true}
+          maxCount={5}
+          counterMode="word"
+          defaultValue="one two three four five"
+        />
+      );
+
+      const clipboardEvent = new Event('paste', {
+        bubbles: true,
+        cancelable: true,
+        composed: true,
+      });
+
+      clipboardEvent.clipboardData = {
+        getData: () => 'one two three four five six seven eight nine ten',
+      };
```

Comment:
is this necessary? doesn't seem like it's being used

Reply:
let me check this, I know I had one working event and one random one

Reply:
yep you are right its not used